bench/wtperf/config.c

```diff
@@ -653,6 +674,45 @@ config_sanity(CONFIG *cfg)
 }
 
 /*
+ * config_to_file --
+ *	Write the final config used in this execution to a file.
+ */
+void
+config_to_file(CONFIG *cfg)
+{
+	CONFIG_QUEUE_ENTRY *config_line;
+	FILE *fp;
+	size_t req_len;
+	char *path;
+
+	fp = NULL;
+
+	/* Backup the config */
+	req_len = strlen(cfg->home) + 100;
+	if ((path = calloc(req_len, 1)) == NULL) {
+		(void)enomem(cfg);
+		goto err;
+	}
+
+	snprintf(path, req_len + 14, "%s/CONFIG.wtperf", cfg->home);
+	if ((fp = fopen(path, "w")) == NULL) {
+		lprintf(cfg, errno, 0, "%s", path);
+		goto err;
+	}
+
+	/* Print the config dump */
+	while (!TAILQ_EMPTY(&cfg->config_head)) {
+		config_line = TAILQ_FIRST(&cfg->config_head);
+		TAILQ_REMOVE(&cfg->config_head, config_line, c);
+		fprintf(fp, "%s\n", config_line->string);
+	}
+err:
+	free(path);
+	if (fp != NULL)
+		(void)fclose(fp);
+}
+
+/*
  * config_print --
  *	Print out the configuration in verbose mode.
```

Comment:
@daveh86 and @agorrod which `config_print` is not part of these changes, it is related for consideration.  That function is only called when verbose > 1 and is an textual style printing of the config.  However, the function isn't really complete AFAICT as there are lots of other config options not printed here.  (I've personally never used it so I don't know if anyone does or how useful they find it.)  Does the new code to write out the config make this obsolete?  If it is useful, it should probably be brought up to date.